protobuf/src/misc.rs

```diff
@@ -10,6 +12,63 @@ pub unsafe fn remove_lifetime_mut<A: ?Sized>(a: &mut A) -> &'static mut A {
     mem::transmute(a)
 }
 
+macro_rules! map_rust_type_to_protobuf {
+    ($rust:path, $protobuf:path) => {
+        impl From<$protobuf> for $rust {
+            fn from(inner: $protobuf) -> Self {
+                inner.value
+            }
+        }
+
+        impl From<$rust> for $protobuf {
+            fn from(inner: $rust) -> Self {
+                let mut value = Self::new();
+                value.value = inner;
+                value
+            }
+        }
+    };
+}
+
+// Map some of well known types to Rust equivalents.
+
+map_rust_type_to_protobuf! { bool, well_known_types::BoolValue }
+map_rust_type_to_protobuf! { Vec<u8>, well_known_types::BytesValue }
+map_rust_type_to_protobuf! { f64, well_known_types::DoubleValue }
+map_rust_type_to_protobuf! { f32, well_known_types::FloatValue }
+map_rust_type_to_protobuf! { i32, well_known_types::Int32Value }
+map_rust_type_to_protobuf! { i64, well_known_types::Int64Value }
+map_rust_type_to_protobuf! { u32, well_known_types::UInt32Value }
+map_rust_type_to_protobuf! { u64, well_known_types::UInt64Value }
+map_rust_type_to_protobuf! { String, well_known_types::StringValue }
+
+impl From<well_known_types::Duration> for Duration {
+    fn from(inner: well_known_types::Duration) -> Self {
+        Duration::new(inner.seconds as u64, inner.nanos as u32)
```

Comment:
This is incorrect at least without explanation: `well_known_types::Duration` can be negative.